### PR TITLE
Docker setup

### LINKED ISSUE #21: A docker container for me, you, and Ruby
#### Issue Summary

It is really nice to be able to develop locally inside a docker container!  Please dockerize this repo! 
- You can probably use the [Dockerfile for sendgrid-ruby](https://github.com/sendgrid/sendgrid-ruby/pull/181)
- Please setup docker-compose to use a local volume for the library files (this is currently different from sendgrid-ruby)
- Please add information to the README and USAGE files about how to startup the container

### PR DESCRIPTION
**Pull request by computermode**
# Resolves #21

### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation about the functionality in the appropriate .md file
- [ ] I have added in line documentation to the code I modified

### Short description of what this PR does:
- Adds a docker file and docker-compose file for the application
- Adds documentation for running the app in a docker container and running the example code within the container

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/smtpapi-ruby?pullRequest=47) <br/>All committers have signed the CLA.

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/smtpapi-ruby?pullRequest=47) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/smtpapi-ruby?pullRequest=47) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/smtpapi-ruby?pullRequest=47) it.</sub>

**Comment by thinkingserious** [comment]
@rnkaufman

We can’t merge your PR without the CLA being signed. 

[ ] **Sign the CLA before November 3rd**
[ ] Tag @mbernier and @thinkingserious to let us know it is signed 

Then we will comment with a link to get your shirt!

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Docker image for smtpapi-ruby
+Running smtpapi-ruby in a Docker container for testing.
+
+## Quickstart
+1. [Install Docker](https://docs.docker.com/engine/installation/) and [docker-compose](https://docs.docker.com/compose/install/) on your machine.
+2. Put your code in `./src`
+3. Run `docker-compose run mock-server`
```
Comment (thinkingserious) [inline docker/README.md:7]: This needs to be run inside the `./docker` directory

Comment (thinkingserious) [reply docker/README.md:7]: After running this, I was put into this prompt: `irb(main):001:0>`

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+4. Run your code with `ruby your_code.rb` inside the container
+
+## Examples
+1. Run `docker-compose run mock-server`.
+2. In your container, run ``ruby examples/example.rb``
```
Comment (thinkingserious) [inline docker/README.md:12]: What should be the expected output here?

File: docker/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Docker image for smtpapi-ruby
+Running smtpapi-ruby in a Docker container for testing.
+
+## Quickstart
+1. [Install Docker](https://docs.docker.com/engine/installation/) and [docker-compose](https://docs.docker.com/compose/install/) on your machine.
+2. Put your code in `./src`
```
Comment (thinkingserious) [inline docker/README.md:6]: I think this step should be clarified a bit with an example.

File: docker/README.md (hunk 2 above)
Comment (thinkingserious) [reply docker/README.md:12]: We should include instruction on how to gracefully shut down.

**Review by thinkingserious — COMMENTED** [review]
Thank you for the PR and my apologies for the delayed review!

**Comment by thinkingserious** [comment]
Closing until PR feedback is addressed.

**Pull request closed without merging**